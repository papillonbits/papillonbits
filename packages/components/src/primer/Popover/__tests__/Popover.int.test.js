import {
  custom,
  regular,
  regularLarge,
  bottom,
  bottomRight,
  bottomLeft,
  left,
  leftBottom,
  leftTop,
  right,
  rightBottom,
  rightTop,
  topLeft,
  topRight,
} from './Popover.int.story'

describe('<Popover />', () => {
  const introOnClickMockFn = jest.fn()
  const acknowledgeOnClickMockFn = jest.fn()

  describe('Event', () => {
    test('must call introOnClick()', () => {
      const mountCustomRender = global.renderMount(custom(introOnClickMockFn, acknowledgeOnClickMockFn))
      mountCustomRender.find('button').first().simulate('click')
      expect(mountCustomRender.props().introOnClick).toBe(introOnClickMockFn)
      expect(mountCustomRender.props().introOnClick).toHaveBeenCalled()
    })

    test('must call acknowledgeOnClick()', () => {
      const mountCustomRender = global.renderMount(custom(introOnClickMockFn, acknowledgeOnClickMockFn))
      mountCustomRender.find('button').last().simulate('click')
      expect(mountCustomRender.props().acknowledgeOnClick).toBe(acknowledgeOnClickMockFn)
      expect(mountCustomRender.props().acknowledgeOnClick).toHaveBeenCalled()
    })
  })

  describe('Render', () => {
    test('must match regular()', () => {
      expect(global.renderToJSON(regular())).toMatchSnapshot()
    })

    test('must match regularLarge()', () => {
      expect(global.renderToJSON(regularLarge())).toMatchSnapshot()
    })

    test('must match bottom()', () => {
      expect(global.renderToJSON(bottom())).toMatchSnapshot()
    })

    test('must match bottomRight()', () => {
      expect(global.renderToJSON(bottomRight())).toMatchSnapshot()
    })

    test('must match bottomLeft()', () => {
      expect(global.renderToJSON(bottomLeft())).toMatchSnapshot()
    })

    test('must match left()', () => {
      expect(global.renderToJSON(left())).toMatchSnapshot()
    })

    test('must match leftBottom()', () => {
      expect(global.renderToJSON(leftBottom())).toMatchSnapshot()
    })

    test('must match leftTop()', () => {
      expect(global.renderToJSON(leftTop())).toMatchSnapshot()
    })

    test('must match right()', () => {
      expect(global.renderToJSON(right())).toMatchSnapshot()
    })

    test('must match rightBottom()', () => {
      expect(global.renderToJSON(rightBottom())).toMatchSnapshot()
    })

    test('must match rightTop()', () => {
      expect(global.renderToJSON(rightTop())).toMatchSnapshot()
    })

    test('must match topLeft()', () => {
      expect(global.renderToJSON(topLeft())).toMatchSnapshot()
    })

    test('must match topRight()', () => {
      expect(global.renderToJSON(topRight())).toMatchSnapshot()
    })
  })
})
