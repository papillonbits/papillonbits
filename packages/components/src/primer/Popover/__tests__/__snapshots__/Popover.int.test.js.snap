// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Popover /> Render must match bottom() 1`] = `
<div
  className="position-relative text-center"
>
  <div
    className={
      Object {
        "pl-2": false,
        "right-0 left-0": false,
      }
    }
  >
    <div
      className="Popover-message text-left p-4 Box box-shadow-large Popover-message--bottom mx-auto mb-2"
    >
      <h4
        className="mb-2"
      >
        Popover heading
      </h4>
      <p>
        Message about this particular piece of UI.
      </p>
      <button
        autoFocus={false}
        className="mt-2 text-bold btn btn-outline"
        data-test={null}
        id={null}
        onClick={[Function]}
        type="button"
      >
        Got it!
      </button>
    </div>
  </div>
  <button
    autoFocus={false}
    className="btn btn-primary"
    data-test={null}
    id={null}
    onClick={[Function]}
    type="button"
  >
    UI
  </button>
</div>
`;

exports[`<Popover /> Render must match bottomLeft() 1`] = `
<div
  className="position-relative"
>
  <div
    className={
      Object {
        "pl-2": true,
        "right-0 left-0": false,
      }
    }
  >
    <div
      className="Popover-message text-left p-4 Box box-shadow-large Popover-message--bottom-left mb-2"
    >
      <h4
        className="mb-2"
      >
        Popover heading
      </h4>
      <p>
        Message about this particular piece of UI.
      </p>
      <button
        autoFocus={false}
        className="mt-2 text-bold btn btn-outline"
        data-test={null}
        id={null}
        onClick={[Function]}
        type="button"
      >
        Got it!
      </button>
    </div>
  </div>
  <button
    autoFocus={false}
    className="btn btn-primary"
    data-test={null}
    id={null}
    onClick={[Function]}
    type="button"
  >
    UI
  </button>
</div>
`;

exports[`<Popover /> Render must match bottomRight() 1`] = `
<div
  className="position-relative text-right pr-2"
>
  <div
    className={
      Object {
        "pl-2": false,
        "right-0 left-0": false,
      }
    }
  >
    <div
      className="Popover-message text-left p-4 Box box-shadow-large Popover-message--bottom-right mb-2"
    >
      <h4
        className="mb-2"
      >
        Popover heading
      </h4>
      <p>
        Message about this particular piece of UI.
      </p>
      <button
        autoFocus={false}
        className="mt-2 text-bold btn btn-outline"
        data-test={null}
        id={null}
        onClick={[Function]}
        type="button"
      >
        Got it!
      </button>
    </div>
  </div>
  <button
    autoFocus={false}
    className="btn btn-primary"
    data-test={null}
    id={null}
    onClick={[Function]}
    type="button"
  >
    UI
  </button>
</div>
`;

exports[`<Popover /> Render must match left() 1`] = `
<div
  className="position-relative"
>
  <div
    className={
      Object {
        "pl-2": false,
        "right-0 left-0": false,
      }
    }
  >
    <div
      className="Popover-message text-left p-4 Box box-shadow-large Popover-message--left"
    >
      <h4
        className="mb-2"
      >
        Popover heading
      </h4>
      <p>
        Message about this particular piece of UI.
      </p>
      <button
        autoFocus={false}
        className="mt-2 text-bold btn btn-outline"
        data-test={null}
        id={null}
        onClick={[Function]}
        type="button"
      >
        Got it!
      </button>
    </div>
  </div>
</div>
`;

exports[`<Popover /> Render must match leftBottom() 1`] = `
<div
  className="position-relative"
>
  <div
    className={
      Object {
        "pl-2": false,
        "right-0 left-0": false,
      }
    }
  >
    <div
      className="Popover-message text-left p-4 Box box-shadow-large Popover-message--left-bottom"
    >
      <h4
        className="mb-2"
      >
        Popover heading
      </h4>
      <p>
        Message about this particular piece of UI.
      </p>
      <button
        autoFocus={false}
        className="mt-2 text-bold btn btn-outline"
        data-test={null}
        id={null}
        onClick={[Function]}
        type="button"
      >
        Got it!
      </button>
    </div>
  </div>
</div>
`;

exports[`<Popover /> Render must match leftTop() 1`] = `
<div
  className="position-relative"
>
  <div
    className={
      Object {
        "pl-2": false,
        "right-0 left-0": false,
      }
    }
  >
    <div
      className="Popover-message text-left p-4 Box box-shadow-large Popover-message--left-top"
    >
      <h4
        className="mb-2"
      >
        Popover heading
      </h4>
      <p>
        Message about this particular piece of UI.
      </p>
      <button
        autoFocus={false}
        className="mt-2 text-bold btn btn-outline"
        data-test={null}
        id={null}
        onClick={[Function]}
        type="button"
      >
        Got it!
      </button>
    </div>
  </div>
</div>
`;

exports[`<Popover /> Render must match regular() 1`] = `
<div
  className="position-relative text-center"
>
  <button
    autoFocus={false}
    className="btn btn-primary"
    data-test={null}
    id={null}
    onClick={[Function]}
    type="button"
  >
    UI
  </button>
  <div
    className={
      Object {
        "pl-2": false,
        "right-0 left-0": true,
      }
    }
  >
    <div
      className="Popover-message text-left p-4 Box box-shadow-large mx-auto mt-2"
    >
      <h4
        className="mb-2"
      >
        Popover heading
      </h4>
      <p>
        Message about this particular piece of UI.
      </p>
      <button
        autoFocus={false}
        className="mt-2 text-bold btn btn-outline"
        data-test={null}
        id={null}
        onClick={[Function]}
        type="button"
      >
        Got it!
      </button>
    </div>
  </div>
</div>
`;

exports[`<Popover /> Render must match regularLarge() 1`] = `
<div
  className="position-relative text-center"
>
  <button
    autoFocus={false}
    className="btn btn-primary"
    data-test={null}
    id={null}
    onClick={[Function]}
    type="button"
  >
    UI
  </button>
  <div
    className={
      Object {
        "pl-2": false,
        "right-0 left-0": true,
      }
    }
  >
    <div
      className="Popover-message text-left p-4 Box box-shadow-large mx-auto mt-2 Popover-message--large"
    >
      <h4
        className="mb-2"
      >
        Popover heading
      </h4>
      <p>
        Message about this particular piece of UI.
      </p>
      <button
        autoFocus={false}
        className="mt-2 text-bold btn btn-outline"
        data-test={null}
        id={null}
        onClick={[Function]}
        type="button"
      >
        Got it!
      </button>
    </div>
  </div>
</div>
`;

exports[`<Popover /> Render must match right() 1`] = `
<div
  className="position-relative"
>
  <div
    className={
      Object {
        "pl-2": false,
        "right-0 left-0": false,
      }
    }
  >
    <div
      className="Popover-message text-left p-4 Box box-shadow-large Popover-message--right"
    >
      <h4
        className="mb-2"
      >
        Popover heading
      </h4>
      <p>
        Message about this particular piece of UI.
      </p>
      <button
        autoFocus={false}
        className="mt-2 text-bold btn btn-outline"
        data-test={null}
        id={null}
        onClick={[Function]}
        type="button"
      >
        Got it!
      </button>
    </div>
  </div>
</div>
`;

exports[`<Popover /> Render must match rightBottom() 1`] = `
<div
  className="position-relative"
>
  <div
    className={
      Object {
        "pl-2": false,
        "right-0 left-0": false,
      }
    }
  >
    <div
      className="Popover-message text-left p-4 Box box-shadow-large Popover-message--right-bottom"
    >
      <h4
        className="mb-2"
      >
        Popover heading
      </h4>
      <p>
        Message about this particular piece of UI.
      </p>
      <button
        autoFocus={false}
        className="mt-2 text-bold btn btn-outline"
        data-test={null}
        id={null}
        onClick={[Function]}
        type="button"
      >
        Got it!
      </button>
    </div>
  </div>
</div>
`;

exports[`<Popover /> Render must match rightTop() 1`] = `
<div
  className="position-relative"
>
  <div
    className={
      Object {
        "pl-2": false,
        "right-0 left-0": false,
      }
    }
  >
    <div
      className="Popover-message text-left p-4 Box box-shadow-large Popover-message--right-top"
    >
      <h4
        className="mb-2"
      >
        Popover heading
      </h4>
      <p>
        Message about this particular piece of UI.
      </p>
      <button
        autoFocus={false}
        className="mt-2 text-bold btn btn-outline"
        data-test={null}
        id={null}
        onClick={[Function]}
        type="button"
      >
        Got it!
      </button>
    </div>
  </div>
</div>
`;

exports[`<Popover /> Render must match topLeft() 1`] = `
<div
  className="position-relative"
>
  <div
    className={
      Object {
        "pl-2": false,
        "right-0 left-0": false,
      }
    }
  >
    <div
      className="Popover-message text-left p-4 Box box-shadow-large Popover-message--top-left"
    >
      <h4
        className="mb-2"
      >
        Popover heading
      </h4>
      <p>
        Message about this particular piece of UI.
      </p>
      <button
        autoFocus={false}
        className="mt-2 text-bold btn btn-outline"
        data-test={null}
        id={null}
        onClick={[Function]}
        type="button"
      >
        Got it!
      </button>
    </div>
  </div>
</div>
`;

exports[`<Popover /> Render must match topRight() 1`] = `
<div
  className="position-relative text-center"
>
  <button
    autoFocus={false}
    className="btn btn-primary"
    data-test={null}
    id={null}
    onClick={[Function]}
    type="button"
  >
    UI
  </button>
  <div
    className={
      Object {
        "pl-2": false,
        "right-0 left-0": true,
      }
    }
  >
    <div
      className="Popover-message text-left p-4 Box box-shadow-large mx-auto mt-2"
    >
      <h4
        className="mb-2"
      >
        Popover heading
      </h4>
      <p>
        Message about this particular piece of UI.
      </p>
      <button
        autoFocus={false}
        className="mt-2 text-bold btn btn-outline"
        data-test={null}
        id={null}
        onClick={[Function]}
        type="button"
      >
        Got it!
      </button>
    </div>
  </div>
</div>
`;
